### PR TITLE
MlKnn: Enable NaNs in training set

### PR DESCRIPTION
**Pull request by c3-robertyoung**
The goal of this implementation is to allow `y` to contain `np.nan` wherever the training data hasn't been specified, i.e. data for which not all labels are positive or negative, some are unknown.

### CODE REVIEW

File: skmultilearn/adapt/mlknn.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ def _compute_cond(self, X, y):
             for label in range(self._num_labels):
```
Comment (c3-robertyoung) [inline skmultilearn/adapt/mlknn.py:188]: 1. make sure the definition of `deltas` still works with nans
2. figure out how to fork repo / use this version in our code
3. make sure label with all nans still works
4. make sure lil matrix format will work with nans

File: skmultilearn/adapt/mlknn.py [hunk 2]
```diff
@@ -145,7 +145,7 @@ def _compute_prior(self, y):
             the prior probability given false
         """
         prior_prob_true = np.array(
-            (self.s + y.sum(axis=0)) / (self.s * 2 + self._num_instances)
+            (self.s + y.sum(axis=0)) / (self.s * 2 + y.isfinite().sum(axis=0))
```
Comment (c3-suvanshdutta) [inline skmultilearn/adapt/mlknn.py:148]: `isfinite` is not a method on a scipy sparse matrix, on line 230, `y` is converted to a lil matrix.
My feeling is that we can probably get away with keeping this as a dense matrix (4000 x 31).  `y` is converted to `label_info`, which is used to access instance-wise ground truth and neighbors' labels

File: skmultilearn/adapt/mlknn.py (hunk 1 above)
Comment (c3-suvanshdutta) [reply skmultilearn/adapt/mlknn.py:188]: 1. `deltas` represents the label-wise sum of 1's for the neighbors of an instance. I think this will work with nans. I can discuss this more tomorrow.
2. todo
3. dry run - should work. I think this is a rare case tbf
4. lil should work with nans, although I recommend using dense arrays wherever numpy apis like isfinite or nansum are needed
Can discuss further tomorrow, and discuss plan/experiments to test this approach

Comment (c3-robertyoung) [reply skmultilearn/adapt/mlknn.py:188]: re 2, check how genAI uses colbert maybe
